addons/im_livechat/models/im_livechat_channel.py

```diff
@@ -58,14 +63,64 @@ def _default_default_message(self):
     chatbot_script_count = fields.Integer(string='Number of Chatbot', compute='_compute_chatbot_script_count')
     rule_ids = fields.One2many('im_livechat.channel.rule', 'channel_id', 'Rules')
 
+    _concurrent_session_greater_than_zero = models.Constraint(
+        "CHECK(concurrent_session_number > 0)",
+        "Concurrent session number should be greater than zero.",
+    )
+
     def _are_you_inside(self):
         for channel in self:
             channel.are_you_inside = self.env.user in channel.user_ids
 
-    @api.depends('user_ids.im_status')
+    @api.depends(
+        "user_ids.im_status",
+        "channel_ids.livechat_active",
+        "channel_ids.last_interest_dt",
+        "channel_ids.rtc_session_ids",
+    )
     def _compute_available_operator_ids(self):
+        session_count_by_operator = {}
+        operators_in_call = self.env["res.partner"]
+        if limited_channel := self.filtered(lambda c: c.concurrent_session == "limited"):
+            old_livechat_dt = fields.Datetime.now() - timedelta(minutes=30)
+            available_user_operators = limited_channel.user_ids.filtered(
+                lambda user: user._is_user_available()
+            )
+            session_count_by_operator = dict(
+                self.env["discuss.channel"]._read_group(
+                    expression.AND(
+                        [
+                            [("livechat_operator_id", "in", available_user_operators.partner_id.ids)],
+                            [("channel_type", "=", "livechat")],
+                            expression.OR(
+                                [
+                                    [("livechat_active", "=", True)],
+                                    [("last_interest_dt", "<=", old_livechat_dt)],
```

Comment:
This condition is wrong anyway, I added a test and edited the domain.
The thing is that we want to count session that are active AND not inactive since 30 minutes.